Source/WebCore/page/Page.cpp

```diff
@@ -5052,6 +5052,22 @@ void Page::performOpportunisticallyScheduledTasks(MonotonicTime deadline)
     if (m_opportunisticTaskScheduler->hasImminentlyScheduledWork())
         options.add(JSC::VM::SchedulerOptions::HasImminentlyScheduledWork);
     commonVM().performOpportunisticallyScheduledTasks(deadline, options);
+
+    deleteRemovedChildNodes();
+}
+
+void Page::deleteRemovedChildNodes()
+{
+    auto* localMainFrame = dynamicDowncast<LocalFrame>(mainFrame());
+    RefPtr document = localMainFrame ? localMainFrame->document() : nullptr;
```

Comment:
I think it's cleaner to have an early exist for when localMainFrame was nullptr.